modules/examiner/php/NDB_Form_examiner.class.inc

```diff
@@ -0,0 +1,381 @@
+        $userCenter   = $user->getCenterID();
+
+        $certification    = $config->getSetting('Certification');
+        $useCertification = $certification['EnableCertification'];
+
+        if ($useCertification) {
+            $centerID = $DB->pselectOne(
+                "SELECT centerID 
+                 FROM examiners
+                 WHERE examinerID=:EID",
+                array('EID' => $this->identifier)
+            );
+
+            return $user->hasPermission('examiner_view')
+                && (($user->hasPermission('examiner_multisite')
+                || $user->getData('CenterID') == $centerID));
+        }
+        return false;
+    }
+
+    /**
+     * Fetches default certification form values from the database.
+     * Creates the certification history table ("Change Log").
+     *
+     * @return array of default form values
+     */
+    function _getDefaults()
+    {
+        $DB = Database::singleton();
+
+        // get the certification results for the given examiner
+        $result = $DB->pselect(
+            "SELECT testID, pass, date_cert, comment 
+             FROM certification
+             WHERE examinerID=:EID",
+            array('EID' => $this->identifier)
+        );
+
+        // set the form defaults for the page
+        foreach ($result as $row) {
+            $defaults['date_cert[' . $row['testID'] . ']'] = $row['date_cert'];
+            $defaults['pass['      . $row['testID'] . ']'] = $row['pass'];
+            $defaults['comment['   . $row['testID'] . ']'] = $row['comment'];
+        }
+
+        return $defaults;
+    }
+
+    /**
+     * Processes the certification form values and inserts updates into the database
+     *
+     * @param array $values Certification form values
+     *
+     * @return nothing
+     */
+    function _process($values)
+    {
+        $DB = Database::singleton();
+
+        foreach ($values['pass'] as $testID => $pass) {
+
+            $date      = $values['date_cert'][$testID];
+            $comment   = trim($values['comment'][$testID]);
+            $date_cert = sprintf(
+                "%04d-%02d-%02d",
+                $date['Y'],
+                $date['M'],
+                $date['d']
+            );
+
+            // get the examinerID - edit_examiner passes the ID through
+            // the identifier, training passes it as a value
+            if (!empty($values['examiner'])) {
+                $examinerID = $values['examiner'];
+            } else {
+                $examinerID = $this->identifier;
+            }
+
+            // Get the certificationID if it exists
+            $certID = $DB->pselectOne(
+                "SELECT certID 
+                 FROM certification
+                 WHERE examinerID=:EID AND testID=:TID",
+                array(
+                 'EID' => $examinerID,
+                 'TID' => $testID,
+                )
+            );
+
+             // if certification for new instrument for the examiner
```

Comment:
It might be cleaner to use $DB->replace(), but that's probably better as a future cleanup of this code.